### PR TITLE
Chore/clean outdated files

### PR DESCRIPTION
**Pull request by JTaeuber**
## Motivation
There were a few outdated files that were pretty much unusable before. This pr aims to clean up a bunch of outdated things and format the code accordingly.

## Changes
- Overhauled pre-commit-config.yaml to be usable and useful
- Adjusted Makefile to have correct information
- Fixed a bunch of python errors pointed out by pytest and mypy
- Formatted all files
- Fixed a few typos and changed install command to be less confusing
- Added correct version and maintainer info
- fixed --interval documentation value mentioned in https://github.com/caas-team/py-kube-downscaler/pull/109

## Tests done
pytest and pre-commit ran with no errors

## TODO

- [x] I've assigned myself to this PR
- [x] It needs to be tested that I didn't break any functionality

### CODE REVIEW

**Comment by JTaeuber** [comment]
@samuel-esp It would be awesome if you could test that i didn't break any functionality. At the moment I cannot do this effectively. I know it's a lot of changes so take your time.

File: Makefile
Comment (jonathan-mayer) [inline Makefile:1]: do we really even need the makefile, the pipelines dont use it and i doubt any one else is

File: kube_downscaler/__init__.py
Comment (jonathan-mayer) [inline kube_downscaler/__init__.py:1]: will this actually still be replaced?

File: pyproject.toml [hunk 1]
```diff
@@ -1,24 +1,22 @@
 [tool]
 [tool.poetry]
 name = "kube-downscaler"
-version = "2020.1.1dev0"
+version = "24.11.0"
 description = "Scale down Kubernetes deployments after work hours"
-authors = ["Henning Jacobs <henning@jacobs1.de>"]
+authors = ["Jonathan Mayer <jonathan.mayer@telekom.de>", "Jan Taeuber <jan.taeuber@telekom.de>", "Johannes Thiem <johannesthiem6@gmail.com>"]
```
Comment (jonathan-mayer) [inline pyproject.toml:6]: @samuel-esp if you want you can add yourself here aswell

**Review by jonathan-mayer — COMMENTED** [review]
mostly LGTM, but we should still test it

File: Makefile
Comment (JTaeuber) [reply Makefile:1]: You have a point. I'll remove it.

**Review by jonathan-mayer — APPROVED** [review]
Was tested :+1:

**Review by samuel-esp — APPROVED** [review]

**Comment by samuel-esp** [comment]
Tested on my side as well and no regressions were found

```
2024-12-02 13:05:04,174 INFO: Downscaler vdev started with admission_controller=, api_server_timeout=10, debug=False, default_downtime=never, default_uptime=Mon-Fri 10:00-12:00 Europe/Rome, deployment_time_annotation=None, downscale_period=never, downtime_replicas=0, dry_run=False, enable_events=False, exclude_deployments=py-kube-downscaler,kube-downscaler,downscaler, exclude_namespaces=py-kube-downscaler,kube-downscaler,kube-system, grace_period=0, include_resources=daemonsets,poddisruptionbudgets,cronjobs,jobs,deployments,statefulsets,scaledobjects, interval=60, matching_labels=, namespace=, once=False, upscale_period=never, upscale_target_only=False
2024-12-02 13:05:04,381 INFO: Scaling down Deployment default/my-deployment from 1 to 0 replicas (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)
2024-12-02 13:05:04,407 INFO: Scaling down Deployment local-path-storage/local-path-provisioner from 1 to 0 replicas (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)
2024-12-02 13:05:04,483 INFO: Suspending CronJob default/my-cronjob (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)
2024-12-02 13:05:04,607 INFO: Pausing ScaledObject default/cron-scaledobject (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)
2024-12-02 13:05:04,911 INFO: Suspending DaemonSet default/my-daemonset (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)
2024-12-02 13:05:05,038 INFO: Scaling down PodDisruptionBudget default/my-pdb from 2 to 0 minAvailable (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)
2024-12-02 13:05:05,256 INFO: Suspending Job default/my-job (uptime: Mon-Fri 10:00-12:00 Europe/Rome, downtime: never)

...

2024-12-02 13:06:12,542 INFO: Downscaler vdev started with admission_controller=, api_server_timeout=10, debug=False, default_downtime=never, default_uptime=Mon-Fri 10:00-15:00 Europe/Rome, deployment_time_annotation=None, downscale_period=never, downtime_replicas=0, dry_run=False, enable_events=False, exclude_deployments=py-kube-downscaler,kube-downscaler,downscaler, exclude_namespaces=py-kube-downscaler,kube-downscaler,kube-system, grace_period=0, include_resources=daemonsets,poddisruptionbudgets,cronjobs,jobs,deployments,statefulsets,scaledobjects, interval=60, matching_labels=, namespace=, once=False, upscale_period=never, upscale_target_only=False
2024-12-02 13:06:12,712 INFO: Scaling up Deployment default/my-deployment from 0 to 1 replicas (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
2024-12-02 13:06:12,737 INFO: Scaling up Deployment local-path-storage/local-path-provisioner from 0 to 1 replicas (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
2024-12-02 13:06:12,872 INFO: Unsuspending CronJob default/my-cronjob (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
2024-12-02 13:06:13,002 INFO: Unpausing ScaledObject default/cron-scaledobject (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
2024-12-02 13:06:13,096 INFO: Unsuspending DaemonSet default/my-daemonset (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
2024-12-02 13:06:13,245 INFO: Scaling up PodDisruptionBudget default/my-pdb from 0 to 2 minAvailable (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
2024-12-02 13:06:13,417 INFO: Unsuspending Job default/my-job (uptime: Mon-Fri 10:00-15:00 Europe/Rome, downtime: never)
```

**Pull request merged**